src/article.ts

```diff
@@ -125,55 +127,68 @@ const tweetContent = (tweetId: string, doc: DocumentFragment): Result<string, No
     return new Err(`There was no blockquote element in the tweet with id: ${tweetId}`);
 }
 
-const parseImage = (element: BlockElement): Option<Image> => {
-    const masterAsset = element.assets.find(asset => asset.typeData.isMaster);
-    const { alt, caption, displayCredit, credit } = element.imageTypeData;
+const parseImage = (element: IBlockElement): Option<Image> => {
+    const masterAsset = element.assets.find(asset => asset?.typeData?.isMaster);
+    const { alt = "", caption = "", displayCredit = false, credit = "" } = element.imageTypeData || {};
 
-    return fromNullable(masterAsset).map(asset => ({
+    return new Some({
```

Comment:
I think you're losing the optionality here. It should be closer to the original file and return None if `masterAsset` is null.